### ISSUE TITLE
Problems about reproducing ViT-Base

### ISSUE THREAD

**Issue by leeyegy** [facebookresearch/mae#30]
Thanks for your great work. 
My reproduction result for ViT-B is only 83.3, which is 0.3 lower than the paper result. And I have no idea what may cause this.
My exp totally follows this repo, except the following changes:
a. I use 32 V100 with batch size 128 (32x128 = 4096). The recommended setting of this repo is 64 V100 with batch size 64(64x64=4096). 
b. I didn't use submitit_pretrain.py but directly use main_pretrain.py for multi-node training.

Are these two changes  the potential causes for performance gap?
Any suggestions would be deeply appreciated~

**Comment by Dongshengjiang** [comment]
I have the same results as you.

**Comment by liuxingbin** [comment]
my reprduction result for VIT-B(400e) is only 82.476 (due to computation limitation, I do not run for 1600e). 

My exp setting:
a.2x8gpu, accum_iter=4. batch_size=64.  (effective batch size:2x8x4x64=4096)
b. multi-node training with main_pretrain.py
c.parameters are strictly followed the original code

And I doubtly think the result under 400e is low.
Could you share the result under 400e?

**Comment by junchen14** [comment]
my reproduced results for ViT-base is also 83.3. The inconsistency reason might be from different pretraining steps. 
their reported results are from pretraining 1600 epochs, but I pretrained 800 epochs.

**Comment by leeyegy** [comment]
@liuxingbin hhh, my result (82.3) of 400e is even lower than yours.

**Comment by leeyegy** [comment]
@junchen14 hi there, did you strictly follow the original settings (64 gpus, submitit_pretrain.py for multi-node training)? because my result (83.3) is achieved by training 1600 epochs instead of 800 epochs.

**Comment by junchen14** [comment]
@leeyegy I run on 8 A100 gpus with 800 epochs.

**Comment by leeyegy** [comment]
I finally got the close result (83.5) of 1600 epochs. The key is to use 64 gpus instead of 32 gpus for pretraining.

**Comment by ChenyangLiu** [comment]
I have the same result on ViT-B by 1600 epochs, acc is 83.3%. And on ViT-L, my acc top1 is 85.5% by training 1600 epochs.
Here are my logs:
[base-pretrain.txt](https://github.com/facebookresearch/mae/files/8065596/base-pretrain.txt)
[base-finetune.txt](https://github.com/facebookresearch/mae/files/8065597/base-finetune.txt)
[large-pretrain.txt](https://github.com/facebookresearch/mae/files/8065604/large-pretrain.txt)
[large-finetune.txt](https://github.com/facebookresearch/mae/files/8065605/large-finetune.txt)

**Comment by youngwanLEE** [comment]
@ChenyangLiu 

How were your training settings(e.g., GPUs, batch size, etc)?

**Comment by ChenyangLiu** [comment]
> @ChenyangLiu
> 
> How were your training settings(e.g., GPUs, batch size, etc)?

32A100s for vit-b, 64 A100s for vit-l, same batch size with the original paper.

**Comment by junchen14** [comment]
@ChenyangLiu    thanks for providing your pretraining details,  according to the paper,  training 1600 epoches for ViT-L should give us 85.9% acc, and traning it with 800 epoches gives 85.5% acc.  

so there are  still some inconsistency, right?

**Comment by endernewton** [comment]
I have recently spent cycles to train the model again (with the same batch-size etc), and in the first shot I am getting 83.5 with ViT-B (after 1600-ep of MAE pre-training and default parameters end-to-end for fine-tuning). As far as know, we don't have numbers for the exact setting (e.g. 400-ep) for some of the runs mentioned above.

**Comment by endernewton** [comment]
For reference, I have pre-trained a 400-ep ViT-B model and after fine-tuning the result is 83.0 top-1 accuracy, and 96.4 top-5. This is only changing the epoch number during pre-training, and following the exact same command for fine-tuning.

I am closing this issue now -- there could be some subtle differences in the setup (e.g. Python, PyTorch, NCCL) for results mentioned above, which is beyond the support of the current code base.

**Comment by kalyani7195** [comment]
Could you please share the command you used for 2 node - 8 gpus each set up using main_pretrain.py? thanks for your help!
